File: frontend/src/schemas.ts
export enum VeiculoStatus {
    ativo = "ativo",
    inativo = "inativo",
    vendido = "vendido",
    indisponivel = "indisponivel",
    duplicado = "duplicado",
}
// class VeiculoStatus(enum.Enum):
//     ativo = 'ativo'
//     inativo = 'inativo'
//     vendido = 'vendido'
//     indisponivel = 'indisponivel'
//     duplicado = 'duplicado'

export interface SiteSchema {
    id: number;
    nome: string;
    url: string;
    veiculos?: VeiculoSchema[];
}

export interface VeiculoSchema {
    id: number;
    marca: string;
    modelo: string;
    ano: number;
    url: string;
    titulo: string;
    site: string;
    status: VeiculoStatus;
    created_at: string;
    favorito: boolean;
    cidade: string;
    historicos?: VeiculoHistoricoSchema[];
    imagens?: VeiculoImagemSchema[];
}

export interface VeiculoHistoricoSchema {
    id: number;
    valor: number;
    quilometragem: number;
    descricao: string;
    datahora: string;
    veiculo_id: number;
    created_at: string;
}

export interface VeiculoImagemSchema {
    id: number;
    url: string;
    status: string;
    veiculo_id: number;
    created_at: string;
}

File: finder/services/schemas.py
import json
from datetime import datetime


class BaseSchema:
    def __init__(self, **kwargs):
        for key, value in kwargs.items():
            setattr(self, key, value)

    def to_json(self):
        return json.loads(
            json.dumps(
                self,
                default=lambda o: o.__dict__,
            )
        )


class VeiculoHistoricoSchema(BaseSchema):
    id: int
    valor: float
    quilometragem: int
    descricao: str
    datahora: datetime
    veiculo_id: int


class VeiculoImagemSchema(BaseSchema):
    id: int
    url: str
    veiculo_id: int


class VeiculoSchema(BaseSchema):
    id: int
    marca: str
    modelo: str
    ano: int
    url: str
    titulo: str
    site: str
    status: str = 'ativo'
    favorito: bool = False

    # def __init__(self, **kwargs):
    #     super().__init__(**kwargs)
    #     self.url = self.url.split('?')[0]
